File: backend/models/Company.js
const mongoose = require('mongoose');

const companySchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    unique: true
  },
  companyName: {
    type: String,
    required: [true, 'Nama perusahaan diperlukan'],
    trim: true
  },
  description: {
    type: String,
    required: [true, 'Deskripsi perusahaan diperlukan'],
    maxlength: [1000, 'Deskripsi maksimal 1000 karakter']
  },
  industry: {
    type: String,
    trim: true
  },
  website: {
    type: String,
    trim: true
  },
  phone: {
    type: String,
    trim: true
  },
  address: {
    street: String,
    city: String,
    province: String,
    postalCode: String
  },
  logo: {
    type: String
  },
  companySize: {
    type: String,
    enum: ['1-10', '11-50', '51-200', '201-500', '500+'],
    default: '1-10'
  },
  foundedYear: {
    type: Number
  },
  isVerified: {
    type: Boolean,
    default: false
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
});

// Update timestamp
companySchema.pre('save', function(next) {
  this.updatedAt = Date.now();
  next();
});

module.exports = mongoose.model('Company', companySchema);

File: frontend/lib/models/company.dart
class Company {
  final String id;
  final String companyName;
  final String description;
  final String? logo;
  final String? website;
  final String? industry;
  final String? location;
  final DateTime createdAt;

  Company({
    required this.id,
    required this.companyName,
    required this.description,
    this.logo,
    this.website,
    this.industry,
    this.location,
    required this.createdAt,
  });

  factory Company.fromJson(Map<String, dynamic> json) {
    return Company(
      id: json['_id'],
      companyName: json['companyName'],
      description: json['description'] ?? '',
      logo: json['logo'],
      website: json['website'],
      industry: json['industry'],
      location: json['location'],
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'companyName': companyName,
      'description': description,
      'logo': logo,
      'website': website,
      'industry': industry,
      'location': location,
    };
  }
}
